liblwgeom/lwcollection.c

```diff
@@ -236,7 +236,7 @@ LWCOLLECTION* lwcollection_add_lwgeom(LWCOLLECTION *col, const LWGEOM *geom)
 LWCOLLECTION *
 lwcollection_segmentize2d(const LWCOLLECTION *col, double dist)
 {
-	uint32_t i;
+	int i;
```

Comment:
That maintains another different warning since `col->ngeoms` is signed and `uint32_t` unsigned:
```
lwcollection.c:245:13: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  for (i=0; i<col->ngeoms; i++)
```

Setting it to `int` solves both issues.

Reply:
It looks like it does not solve but instead hides them :)

Can you instead try changing all collections' ngeoms and maxgeoms it to uint32_t (or even size_t) in
https://github.com/postgis/postgis/blame/9715e11745bb54a981c7402d46b4191e9d680a81/liblwgeom/liblwgeom.h.in#L584 ?

Reply:
It was being discussed in here (http://lists.osgeo.org/pipermail/postgis-devel/2017-December/026773.html) and I'm all for it, but I don't want to go ahead and edit hundreds of lines to get it rejected later.